src/mqt/yaqs/digital/utils/qasm_utils.py

```
@@ -0,0 +1,146 @@
+_OPENQASM_VERSION_RE = re.compile(r"OPENQASM\s+(\d+)", re.IGNORECASE)
+_QASM3_IMPORT_MSG = (
+    "OpenQASM 3 loading requires the optional package qiskit-qasm3-import. "
+    "Install with: pip install mqt-yaqs[qasm3] or pip install qiskit[qasm3-import]"
+)
+_INVALID_INPUT_MSG = (
+    "Expected a QuantumCircuit, a filesystem path to an OpenQASM file, "
+    "or a raw OpenQASM string whose first substantive line declares OPENQASM."
+)
+
+
+def _parse_qasm_version(text: str) -> Literal["2", "3"] | None:
```

Ruff (0.15.17)

[warning] Too many branches (13 > 12)

(PLR0912)